### PR TITLE
Refactor conv_transpose padding

### PR DESCRIPTION
**Pull request by guillaumebaquiast**
This PR refactors the way padding arguments are handled by Conv Transpose functions, to fix the misalignment between backends (see https://github.com/keras-team/keras-core/issues/774).

The main change is to create two functions to convert the padding arguments from the Keras convention to the JAX / Torch conventions. As raised in the issue, it seems to me that some padding configurations are not possible to be translated in Torch; in which case, a warning is raised when using those configurations on a Torch backend.

Some caveat / interrogations:
* The padding conventions of conv transpose are not always very well documented by the original backends, so I had to infer their implementation. This PR implements the best of my understanding, but could be missing some edge case.
* I added some unit tests that check that the implementation gives the same result than the original keras implementation on a bunch of inputs. This helped capturing many edge cases, but not sure if you want to have the unit tests depending on tensorflow keras.
* Currently, a warning is raised when the padding configuration lead to differences between Torch and other backend. The warning is only raised when using Torch as backend, so not sure if it is better to raise it for all backend (the use-case would be that if a model is trained with a Torch backend, it wouldn't behave as expected when used with a TF backend).

### CODE REVIEW

File: keras_core/backend/common/backend_utils.py [hunk 1]
```diff
@@ -1,155 +1,253 @@
...
+            f"You might experience inconsistencies accross backends when "
+            f"calling conv transpose with {kernel_size}: kernel_size, "
+            f"{stride}: stride, {dilation_rate}: dilation_rate, "
+            f"{padding}: padding, {output_padding}: output_padding."
         )
-    return length
 
+    if torch_output_padding >= stride:
+        raise ValueError(
+            f"The padding arguments (`padding`={padding}) and "
```
Comment (fchollet) [inline keras_core/backend/common/backend_utils.py:96]: Remove backticks in the warning messages

File: keras_core/backend/common/backend_utils.py [hunk 2]
```diff
@@ -1,155 +1,253 @@
...
+            -((kernel_size % 2 - kernel_size + output_padding) // 2), 0
+        )
+        torch_output_padding = (
+            2 * torch_padding + kernel_size % 2 - kernel_size + output_padding
+        )
+
+    if torch_padding > 0 and torch_output_padding > 0:
+        warnings.warn(
+            f"You might experience inconsistencies accross backends when "
+            f"calling conv transpose with {kernel_size}: kernel_size, "
```
Comment (fchollet) [inline keras_core/backend/common/backend_utils.py:89]: Use `=` instead of `:`

**Comment by fchollet** [comment]
> The warning is only raised when using Torch as backend, so not sure if it is better to raise it for all backend

Raising only in torch is fine since it's the odd one out.

File: keras_core/layers/convolutional/conv_transpose_test.py [hunk 3]
```diff
@@ -749,3 +750,90 @@ def test_conv3d_transpose(
...
+                "`output_padding` greater than `strides` is not supported"
+            )
+
+        input = np.ones(shape=(1, 3, 1))
+        kernel_weights = np.arange(1, kernel_size + 1).reshape(
+            (kernel_size, 1, 1)
+        )
+
+        # TF layer
+        tf_layer = tf.keras.layers.Conv1DTranspose(
```
[the remark this answers is not in the document]
Comment (guillaumebaquiast) [reply keras_core/layers/convolutional/conv_transpose_test.py]: OK, I now used `np_conv1d_transpose` to remove the dependency on tf.keras.
Note that it's not 100% fool-proof, because `np_conv1d_transpose` depends on some internal keras-core function (for instance to get the output shape), so if those are wrong, the test could still pass.

**Review by fchollet — APPROVED** [review]
LGTM, thank you!
